src/Admin/SiteHealth.php

```diff
@@ -0,0 +1,368 @@
+						esc_html__( 'Learn more about persistent object caching', 'amp' ),
+						/* translators: The accessibility text. */
+						esc_html__( '(opens in a new tab)', 'amp' )
+					),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Gets the test result data for whether the curl_multi_* functions exist.
+	 *
+	 * @return array The test data.
+	 */
+	public function curl_multi_functions() {
+		$undefined_curl_functions = array_filter(
+			[
+				'curl_multi_add_handle',
+				'curl_multi_exec',
+				'curl_multi_init',
+			],
+			static function( $function_name ) {
+				return ! function_exists( $function_name );
+			}
+		);
+
+		$data = [
+			'badge'       => [
+				'label' => esc_html__( 'AMP', 'amp' ),
+				'color' => $undefined_curl_functions ? 'orange' : 'green',
+			],
+			'description' => esc_html__( 'The AMP plugin performs better when these functions are available.', 'amp' ),
+			'actions'     => sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				'https://www.php.net/manual/book.curl.php',
+				esc_html__( 'Learn more about these functions', 'amp' ),
+				/* translators: The accessibility text. */
+				esc_html__( '(opens in a new tab)', 'amp' )
+			),
+			'test'        => 'amp_curl_multi_functions',
+		];
+
+		if ( $undefined_curl_functions ) {
+			return array_merge(
+				$data,
+				[
+					'status'      => 'recommended',
+					'label'       => esc_html(
+						_n(
+							'A curl_multi_* function is not defined',
+							'Some curl_multi_* functions are not defined',
+							count( $undefined_curl_functions ),
+							'amp'
+						)
+					),
+					'description' => wp_kses(
+						sprintf(
+							/* translators: %s: the name(s) of the curl_multi_* PHP function(s) */
+							_n(
+								'The following curl_multi_* function is not defined: %s. The AMP plugin performs better when this function is available.',
+								'The following curl_multi_* functions are not defined: %s. The AMP plugin performs better when these functions are available.',
+								count( $undefined_curl_functions ),
+								'amp'
+							),
+							implode(
+								', ',
+								array_map(
+									static function( $function_name ) {
+										return sprintf( '<code>%s()</code>', $function_name );
+									},
+									$undefined_curl_functions
+								)
+							)
+						),
+						[ 'code' => [] ]
+					),
+				]
+			);
+		} else {
+			return array_merge(
+				$data,
+				[
+					'status' => 'good',
+					'label'  => esc_html__( 'The curl_multi_* functions are defined.', 'amp' ),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Gets the test result data for whether the proper ICU version is available.
+	 *
+	 * @return array The test data.
+	 */
+	public function icu_version() {
+		$icu_version       = defined( 'INTL_ICU_VERSION' ) ? (float) INTL_ICU_VERSION : null;
+		$minimum_version   = 4.6;
```

Comment:
Thanks to Brandon Kraft's [comment](https://github.com/ampproject/amp-wp/issues/1439#issuecomment-422074851), it looks like ICU version `4.6` introduced the `INTL_IDNA_VARIANTE_UTS46` constant.

And as long as that constant is defined, it will be passed to `idn_to_utf8()`:

https://github.com/ampproject/amp-wp/blob/39336b78be7a4ba8b3a09a6f43d9f66e3c201e05/includes/class-amp-http.php#L233

...and that won't trigger this deprecation warning [mentioned here](https://github.com/ampproject/amp-wp/issues/2199#issuecomment-557235606):

https://github.com/php/php-src/pull/2355/files#diff-24931a05f9e656989a55c44cf8a72f9eR294

Reply:
Also, I tested this locally with PHP 7.2:

```sh
$ wp shell
$ wp> idn_to_utf8( 'foobar', IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46 );
=> string(6) "foobar" // As expected, no deprecation warning
$ wp> idn_to_utf8( 'foobar', IDNA_DEFAULT, INTL_IDNA_VARIANT_2003 );
PHP Deprecated:  idn_to_utf8(): INTL_IDNA_VARIANT_2003 is deprecated in phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php(46) : eval()'d code on line 1
PHP Stack trace:
PHP   1. {main}() /usr/local/bin/wp:0
....
```

Reply:
So I think the minimum ICU version of `4.6` should be good.